### PR TITLE
Feature/372

### PR DESCRIPTION
**Pull request by ddelpiano**
- Menu grouping implemented
- Configurable from the focusTermConfiguration.js as an array of objects, the label property specify the name of the sub menu, the keys specify the substring to filter queries and include them into the sub menu.
- Fixed a bug where queries were piled up opening 2 or more consecutively.

### CODE REVIEW

File: components/interface/FocusTerm.js [hunk 1]
```diff
@@ -194,12 +256,8 @@ export default class FocusTerm extends React.Component {
       var entity = Model[otherId];
       // clear query builder unless ctrl pressed them add to compound.
       this.props.queryBuilder.open();
-      if (!GEPPETTO.isKeyPressed("shift")) {
```
Comment (gidili) [inline components/interface/FocusTerm.js]: @ddelpiano are we sure this is not needed anymore (the SHIFT check)?

**Comment by Robbie1977** [comment]
Yeah, the shift should allow queries to be added to the old ones.

File: components/interface/FocusTerm.js (hunk 1 above)
Comment (ddelpiano) [reply components/interface/FocusTerm.js]: will fix this shortly

Comment (Robbie1977) [reply components/interface/FocusTerm.js]: Note: Revert on this still needed

**Pull request merged**